sklearn/model_selection/_search.py

```diff
@@ -847,10 +858,16 @@ class GridSearchCV(BaseSearchCV):
             - A string, giving an expression as a function of n_jobs,
               as in '2*n_jobs'
 
-    iid : boolean, default=True
+    iid : boolean, default='warn'
         If True, the data is assumed to be identically distributed across
         the folds, and the loss minimized is the total loss per sample,
```

Comment:
This has always been a weird description. Could we clarify that it is an average score across folds, weighted by the number of samples in each test set...?